jena/src/main/scala/org/w3/banana/jena/io/JenaRDFReader.scala

```diff
@@ -47,18 +51,17 @@ final class TripleSink(implicit ops: JenaOps) extends StreamRDF {
 object JenaRDFReader {
 
   def makeRDFReader[S](ops: JenaOps, lang: Lang): RDFReader[Jena, Try, S] = new RDFReader[Jena, Try, S] {
-    val factory = RDFParserRegistry.getFactory(lang)
 
     def read(is: InputStream, base: String): Try[Jena#Graph] = Try {
       val sink = new TripleSink
-      factory.create(lang).read(is, base, null, sink, null)
-      //      RDFDataMgr.parse(sink, is, base, lang)
+      RDFParser.create().source(is).lang(lang).base(base).parse(sink)
       sink.graph
     }
 
     def read(reader: Reader, base: String): Try[Jena#Graph] = Try {
```

Comment:
This function could be simpler by just creating an InputStream and forward to read(id: InputStream, base: String) in line 55.

Reply:
can you add that fix to one of your PRs?

Reply:
ah yes, though at the same time the sad thing is that that means that one looses all the information that is already contained in the reader - (The reader has already converted all the bytes to chars!) ... I find this reliance on bytestreams by Jena pretty horrible.